container-registry-configuration/README.md

```diff
@@ -0,0 +1,191 @@
+
+#### 2.b Using authenticated accounts
+
+We can pass credentials to `containerd` configuration to be not limited by registry provider limits.
+
+#### Decision
+
+`2.b` is selected.
+We will use authenticated accounts.
+
+
+### 3. Registries
+
+The intranet page (See `Registry Mirrors`) states this as a desired configuration:
+  - `docker.io` as primary public registry since it has a privilege in docker daemon.
+  - Our own Azure Container Registry as secondary registry because of the security concerns
+
+Since we use `containerd`, docker has not a privilege anymore but there is no need to change this order at the moment. We will follow this configuration for CAPI clusters too.
+
+### 4. Accounts
+
+- 4.a Giantswarm Account + Single account
+- 4.b Giantswarm Account + Per MC including its WCs
+- 4.c Giantswarm Account + Per WC
+- 4.d Customer Accounts
+
+#### Decision
+
+`4.b` is selected.
+
+Since it is a part of the platform itself, we are going to use Giant Swarm accounts in containerd configuration.
+As we do for other operators, we will follow "per MC" approach here to.
+
+### 5. How to provide credentials to WCs
+
+#### 5.a cluster-apps-operator
+
+We can propogate credentials from MC to WC by using `cluster-apps-operator`. It can create a secret per WC with the credentials.
+
+#### 5.b Optional interface + Gitops
+
+We can delegate the responsibility of creating credentials secret to the creator of `cluster-$provider` apps. GitOps can be useful to template the secrets per WC.
+
+#### Decision
+
+`5.b` is selected.
+
+We don't want to add another responsility to `cluster-apps-operator`. 
+
+
+### 6. Default configuration for WCs
+
+- 6.a: Configuring WCs by default
+- 6.b: Not configuring WCs by default
+
+#### Decision
+
+`6.a` is selected.
+
+### 7. Where to put the credentials
+
+#### 7.a Customer's Git Repository
+
+We can put the secret into customers' git repositories. Customers will access to our credentials.
+
+#### 7.b Management Cluster Fleet
+
+We can put the secret into our `management-clusters-fleet` repo. We will need to give reference to that secret in customers' git repository. Management of these secrets can be so costly when we need to create the secret in each organization namespace.
+
+#### 7.c Catalog configuration
+
+We can use catalog configurations (See <mc-name>/appcatalog folder in `installations` repo) to provide a MC specific configuration to all charts in the cluster.
+
+#### Decision
+
+7.c is selected.
```

Comment:
Important comment about this solution. Ii totally makes sense short term, but in a long term we might (just might - nothing's set in stone) deprecate support for catalog configs. Why? 
* they are supper-hidden currently; there's no indication in the App CR that a config from a not very related Catalog CR is being used. Yes, we can indicate this in App's status, but still:
* from an unaware user's perspective, they work like magic, in the unreliable, hard to debug way: you install two apps, one gets the config the other doesn't - why? (they come from different catalogs). You move and app between catalogs (or just stop using the `-test` catalog and stuff breaks, even though app's config is exactly the same

In the long run, I feel it would be much better to provide a different source of cross-app config, like a CM/Secret name `cluster-default` (just an idea) that is included in every app's `extraConfigs` by default, with low priority.

Let me know what you think.

Reply:
We picked the catalog configs since we don't have any other mechanism now. Personally, I am not happy with the solution. Now, I am trying to implement it but I need to use `opctl encrypt` and `opsctl decrypt` for catalog configuration in the installations repo, which will be something new for us. 

Regarding `cluster-default` secret, it should be created in every org namespace automatically. There is no operator for it yet.